src/ArborX_LinearBVH.hpp

```diff
@@ -33,18 +33,20 @@ template <typename DeviceType>
 struct TreeVisualization;
 }
 
-template <typename DeviceType>
+template <typename MemorySpace, typename Enable = void>
 class BoundingVolumeHierarchy
 {
 public:
-  using device_type = DeviceType;
+  using memory_space = MemorySpace;
+  static_assert(Kokkos::is_memory_space<MemorySpace>{}, "");
```

Comment:
Why not give a useful message?

Reply:
because I judged
```
error: static_assert failed due to requirement 'Kokkos::is_memory_space<INVALID_TYPE>::value' ""
  static_assert(Kokkos::is_memory_space<MemorySpace>::value, "");
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
was good enough